hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2785,3 +2794,224 @@ func reconcileMachineConfigServerService(svc *corev1.Service) error {
+	role := machineapprover.Role(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, role, func() error {
+		return reconcileMachineApproverRole(role)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role: %w", err)
+	}
+
+	// Reconcile machine-approver service account
+	sa := machineapprover.ServiceAccount(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, sa, NoopReconcile)
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver service account: %w", err)
+	}
+
+	// Reconcile machine-approver role binding
+	rolebinding := machineapprover.RoleBinding(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, rolebinding, func() error {
+		return reconcileMachineApproverRoleBinding(rolebinding, role, sa)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role binding: %w", err)
+	}
+	config := machineapprover.ConfigMap(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, config, func() error {
+		return reconcileMachineApproverConfig(config)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver config: %w", err)
+	}
+
+	// The deployment depends on the kubeconfig being reported.
+	if hcp.Status.KubeConfig != nil {
+		// Resolve the kubeconfig secret for machine-approver
+		kubeconfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: hcp.Namespace,
+				Name:      "service-network-admin-kubeconfig",
+			},
+		}
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(kubeconfigSecret), kubeconfigSecret)
+		if err != nil {
+			return fmt.Errorf("failed to get hosted controlplane kubeconfig secret %q: %w", kubeconfigSecret.Name, err)
+		}
+
+		// Reconcile machine-approver deployment
+		image := imageMachineApprover
+		if _, ok := hcluster.Annotations[hyperv1.MachineApproverImage]; ok {
+			image = hcluster.Annotations[hyperv1.MachineApproverImage]
+		}
+		deployment := machineapprover.Deployment(controlPlaneNamespace.Name)
+		_, err = controllerutil.CreateOrUpdate(ctx, r.Client, deployment, func() error {
+			return reconcileMachineApproverDeployment(deployment, hcluster, sa, kubeconfigSecret, config, image, r.HypershiftOperatorImage)
+		})
+		if err != nil {
+			return fmt.Errorf("failed to reconcile machine-approver deployment: %w", err)
+		}
+	}
+
+	return nil
+}
+
+type ClusterMachineApproverConfig struct {
+	NodeClientCert NodeClientCert `json:"nodeClientCert,omitempty"`
+}
+type NodeClientCert struct {
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+func reconcileMachineApproverConfig(cm *corev1.ConfigMap) error {
+	// Enable the client cert csr approval
+	cfg := ClusterMachineApproverConfig{
+		NodeClientCert: NodeClientCert{
+			Disabled: false,
+		},
+	}
+	cfgJSON, err := json.Marshal(cfg)
+	if err != nil {
+		return err
+	}
+	b, err := yaml.JSONToYAML(cfgJSON)
```

Comment:
can't we just yaml.Marshal(cfg)?

Reply:
https://github.com/openshift/hypershift/blob/main/api/scheme.go#L29